### PR TITLE
Feature/gradle build

### PR DESCRIPTION
**Pull request by raptor22fa**
Liferay boilerplate can be build by one gradle command.

### CODE REVIEW

File: boilerplates/react-union-boilerplate-liferay-basic/tools/amdLoaderScripts.js [hunk 1]
```diff
@@ -26,12 +26,12 @@ const targetDirectory = path.join(
 	appDirectory,
 	'liferay/amd-loader/src/main/resources/META-INF/resources'
 );
-const configFilePath = path.join(targetDirectory, 'config.js');
+const configFilePath = path.join(buildDirectory, 'config.js');
 
 const joinNonEmpty = xs => xs.filter(Boolean).join('.');
 
 const getLiferayFilePath = (appName, hash) =>
-	path.join(targetDirectory, `${joinNonEmpty([appName, hash])}.js`);
```
Comment (wafflepie) [inline boilerplates/react-union-boilerplate-liferay-basic/tools/amdLoaderScripts.js]: You can remove the targetDirectory variable altogether, it's not used anywhere else :)

File: boilerplates/react-union-boilerplate-liferay-basic/liferay/amd-loader/build.gradle [hunk 2]
```diff
@@ -1,38 +1,22 @@
...
-
-repositories {
-	mavenLocal()
-
-	maven {
-		url "https://cdn.lfrs.sl/repository.liferay.com/nexus/content/groups/public"
-	}
+    compileOnly globalDependency.portalKernel
+    compileOnly globalDependency.osgiCmpn
+	compileOnly globalDependency.portalAmdLoader
```
Comment (wafflepie) [inline boilerplates/react-union-boilerplate-liferay-basic/liferay/amd-loader/build.gradle]: Inconsistent indentation

File: boilerplates/react-union-boilerplate-liferay-basic/tools/amdLoaderScripts.js (hunk 1 above)
Comment (raptor22fa) [reply boilerplates/react-union-boilerplate-liferay-basic/tools/amdLoaderScripts.js]: Done

File: boilerplates/react-union-boilerplate-liferay-basic/liferay/amd-loader/build.gradle (hunk 2 above)
Comment (raptor22fa) [reply boilerplates/react-union-boilerplate-liferay-basic/liferay/amd-loader/build.gradle]: Reformatted.

**Comment by wafflepie** [comment]
Thanks a lot!! 🙂

File: boilerplates/react-union-boilerplate-liferay-basic/liferay-theme/build.gradle [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    dependencies {
+        classpath "com.moowork.gradle:gradle-node-plugin:1.2.0"
+    }
+}
+
+apply plugin: 'com.moowork.gulp'
+
+node {
+    version = '8.9.1'
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/liferay-theme/build.gradle]: Different version of node for theme and react union. Is it possible to have just one version for both? If not its ok.

File: boilerplates/react-union-boilerplate-liferay-basic/liferay-theme/build.gradle [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+// makes sure on each build that gulp is installed
+gulp_build.dependsOn 'installGulp'
+
+// processes your package.json before running gulp build
+gulp_build.dependsOn 'npmInstall'
+
+// runs "gulp build" as part of gradle build
+task build(dependsOn: 'gulp_build') {
+}
```
Comment (Juldik) [inline boilerplates/react-union-boilerplate-liferay-basic/liferay-theme/build.gradle]: missing empty line at the end of file.

**Comment by tommmyy** [comment]
@raptor22fa thank you very much for amazing work!

**Pull request merged**